src/software/ai/hl/stp/action/movespin_action_test.cpp

```diff
@@ -33,10 +33,9 @@ TEST(MoveSpinActionTest, robot_at_destination)
     // We call the action twice. The first time the Intent will always be returned to
     // ensure the Robot is doing the right thing. In all future calls, the action will be
     // done and so will return a null pointer
-    auto intent_ptr = action.updateStateAndGetNextIntent(robot, Point(0, 0),
-                                                         AngularVelocity::full(), 0);
-    intent_ptr      = action.updateStateAndGetNextIntent(robot, Point(0, 0),
-                                                    AngularVelocity::full(), 0);
+    action.updateControlParams(robot, Point(0, 0), AngularVelocity::full(), 0);
+    action.getNextIntent();
```

Comment:
Can we test that this is the right value?

Reply:
Hmm, looking across the action tests, this pattern of just calling `getNextIntent` without checking the return value is pretty common (and is equivalent to what we were doing before). If we're going to fix it, I think that's a much bigger job that's out-of-scope for this PR. If you still think this is an issue, could you post in #software and we can have a discussion about whether or not to address this across all action tests? This could definitely be something we might change, just not sure if the effort is justified :man_shrugging:

Reply:
Yea we were previously not looking at the result anyway, we just needed to make multiple calls to advance the coroutine enough to check we were "done"